pkg/listwatch/namespace_denylist.go

```diff
@@ -179,3 +179,10 @@ func newDenylistWatch(l log.Logger, denylist map[string]struct{}, next watch.Int
 
 	return proxy
 }
+
+func getNamespace(obj metav1.Object) string {
```

Comment:
Nit: Could we just add a short function comment for this, mainly mentioning that this ensures filtering out `Namespace` objects correctly. Or rename to `getNamespaceName()` as we are not getting the Namespace object. WDYT?

Reply:
added a comment (: